droidlet/perception/craftassist/low_level_perception.py

```diff
@@ -34,6 +39,43 @@ def __init__(self, agent, perceive_freq=5):
         self.memory = agent.memory  # NOTE: remove this once done
         self.pending_agent_placed_blocks = set()
         self.perceive_freq = perceive_freq
+        self.perception_logger = PerceptionLogger(
+            "vision_error_details.csv", ["command", "action_dict", "time", "perception_error", "world_state"]
+        )
+
+        @sio.on("saveErrorDetailsToCSV")
+        def save_error_details(sid, data):
+            """Save error details to error logs.
+            The fields are
+                ["command", "action_dict", "time", "perception_error", "world_state"]
+            """
+            logging.info("Saving vision error details: %r" % (data))
+            if "msg" not in data:
+                logging.info("Could not save vision error details due to error in dashboard backend.")
+                return
+            is_perception_error = data["perception_error"]
+            if True:
```

Comment:
Should this be `if is_perception_error:`?

Reply:
Yes it is! I did this for testing. Fixed in the latest commit.